src/accounts/forms.py

```diff
@@ -16,24 +17,55 @@ class MyAuthenticationForm(AuthenticationForm):
             'required': True,
         })
     )
+    class Meta:
+        model = User
 
 class MyUserCreationForm(UserCreationForm):  # pylint: disable=too-many-ancestors
+    username = forms.EmailField(
+        required=False,
+        widget=forms.EmailInput(attrs={
+            'class': 'input',
+            'type': 'email'
+        })
+    )
     email = forms.EmailField(
         required=True,
         widget=forms.EmailInput(attrs={
             'class': 'input',
             'type': 'email',
+            'placeholder': 'Email',
+        })
+    )
+    first_name = forms.CharField(
+        required=True,
+        widget=forms.TextInput(attrs={
+            'class': 'input',
+            'type': 'text',
+            'placeholder': 'First Name',
+        })
+    )
+    last_name = forms.CharField(
+        required=True,
+        widget=forms.TextInput(attrs={
+            'class': 'input',
+            'type': 'text',
+            'placeholder': 'Last Name',
         })
     )
     password1 = forms.CharField(
         widget=forms.PasswordInput(attrs={
             'class': 'input',
             'type': 'password',
+            'placeholder': 'Enter Password',
         })
     )
     password2 = forms.CharField(
         widget=forms.PasswordInput(attrs={
             'class': 'input',
             'type': 'password',
+            'placeholder': 'Re-enter Password',
         })
     )
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'first_name', 'last_name', 'password1', 'password2']
```

Comment:
do we ever authenticate on signup that `password1 == password2` ? I feel that this should be done on the front-end to prevent having to send an invalid form to the back-end.

Reply:
This is how Django does it on the back end.

Reply:
nice . makes sense!